queue/src/main/java/com/bazaarvoice/emodb/queue/core/kafka/KafkaProducerService.java

```diff
@@ -0,0 +1,73 @@
+package com.bazaarvoice.emodb.queue.core.kafka;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.concurrent.Future;
+
+public class KafkaProducerService {
+    private static final Logger _log = LoggerFactory.getLogger(KafkaProducerService.class);
+    private final KafkaProducer<String, String> producer; // Changed to String
+
+    public KafkaProducerService() {
+        this.producer = new KafkaProducer<>(KafkaConfig.getProducerProps());
+        _log.info("KafkaProducerService initialized with producer properties: {}", KafkaConfig.getProducerProps());
+    }
+
+    /**
+     * Sends each message from the collection to the specified Kafka topic separately.
+     *
+     * @param topic   The Kafka topic.
+     * @param events  The collection of messages to be sent.
+     */
+    public void sendMessages(String topic, Collection<String> events, String queueType) {
+        _log.info("Sending {} messages to topic '{}'", events.size(), topic);
+        for (String event : events) {
+            _log.debug("Sending message: {}", event);
+            sendMessage(topic, event,queueType);
+        }
+        _log.info("Finished sending messages to topic '{}'", topic);
+    }
+
+    /**
+     * Sends a single message to the specified Kafka topic.
+     *
+     * @param topic   The Kafka topic.
+     * @param message The message to be sent.
+     */
+    public void sendMessage(String topic, String message, String queueType) {
+        ProducerRecord<String, String> record = new ProducerRecord<>(topic, message);
+        _log.debug("Preparing to send message to topic '{}' with value: {}", topic, message);
```

Comment:
**suggestion:** Considering the millions of messages planned to go through this piece of code, that's gonna be a hefty amount of logs, my suggestion would be to log the error only and log info level up in the calling `sendMessages` function for each batch - if this is just temporary during early e2e testing, then add a TODO to show that intent please

Reply:
yes these were temporary logs while e2e testing, removed now to avoid the bombardment of logs

Reply:
Thanks!